src/appleseed.shared/application/progresstilecallback.cpp

```diff
@@ -80,12 +81,19 @@ namespace
 
         void on_tiled_frame_begin(const Frame* frame) override
         {
-            m_stopwatch.start();
+            // Do not restart the stopwatch if it is already running.
+            if (m_stopwatch.get_seconds() <= 0.0)
+                m_stopwatch.start();
         }
 
         void on_tiled_frame_end(const Frame* frame) override
         {
-            m_stopwatch.clear();
+            const size_t total_tiles = frame->image().properties().m_tile_count * m_pass_count;
+
+            // Clear the stopwatch only when rendered tiles exceeds total tiles.
+            // This makes sure that we don't clear the stopwatch in between passes.
+            if (m_rendered_tiles > total_tiles)
```

Comment:
How can this ever happen?

Reply:
It cannot happen, my bad. There should be equality.